cmd/juju/application/config.go

```diff
@@ -306,6 +348,36 @@ func (c *configCommand) getConfig(client configCommandAPI, ctx *cmd.Context) err
 	return c.out.Write(ctx, resultsMap)
 }
 
+// validateValues reads the values provided as args and validates that they are
+// valid UTF-8.
+func (c *configCommand) validateValues(ctx *cmd.Context) (map[string]string, error) {
+	settings := map[string]string{}
+	for k, v := range c.values {
+		//empty string is also valid as a setting value
+		if v == "" {
+			settings[k] = v
+			continue
+		}
+
+		if v[0] != '@' {
+			if !utf8.ValidString(v) {
+				return nil, errors.Errorf("value for option %q contains non-UTF-8 sequences", k)
+			}
+			settings[k] = v
+			continue
+		}
+		nv, err := readValue(ctx, v[1:])
+		if err != nil {
+			return nil, err
```

Comment:
I just moved that below where it was used because that is someone else's nit:) I didn't read through the existing code carefully though. Nice catch:)